src/data_snack/connections/mongo.py

```diff
@@ -20,53 +28,64 @@ def _get_entity_collection(self, entity_type: Type[Entity]) -> pymongo.collectio
         )
         return collection
 
-    def get(self, entity_type: Type[Entity], key: str) -> Optional[bytes]:
-        collection = self._get_entity_collection(entity_type)
-        result = collection.find_one({"_id": key})
+    def get(self, key: Key) -> Optional[bytes]:
+        collection = self._get_entity_collection(key.entity_type)
+        result = collection.find_one({"_id": key.keystring})
         if not result:
             return None
         del result['_id']
-        return result
+        return result  # TODO: wrong typing - investigate
+
+    @validate_keys
+    def get_many(self, keys: List[Key], /) -> Dict[str, Optional[Dict]]:
+        if not (entity_type := getattr(next(iter(keys), None), "entity_type", None)):
```

Comment:
I would consider moving it to some `utils` and or at the beginning of the file and calling it `_get_first_element_field_or_none` :shrug:

Reply:
Moved to utils since reused in another place (unit tests added).